### PR TITLE
Hm 3

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

**Review by andromash — APPROVED** [review]
Good job!

File: src/main/java/com/internet/shop/controllers/IndexController.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class IndexController extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/IndexController.java]: this code line is longer than 100 chars

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<%--
+  Created by IntelliJ IDEA.
+  User: 38093
+  Date: 08.09.2020
+  Time: 12:10
+  To change this template use File | Settings | File Templates.
+--%>
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/views/index.jsp]: don't forget about checklist
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_15.html

**Review by RomanDubovskyi — COMMENTED** [review]
let's connect travis to our project, so it will take care of codestyle checks

File: src/main/java/com/internet/shop/controllers/IndexController.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/IndexController.java]: Done.

File: src/main/webapp/WEB-INF/views/index.jsp (hunk 2 above)
Comment (AndrewPolishchuc) [reply src/main/webapp/WEB-INF/views/index.jsp]: Sure. Done.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (RomanDubovskyi) [inline .travis.yml]: add empty line to the end of the file

File: src/main/java/com/internet/shop/dao/GenericDao.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.internet.shop.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface GenericDao<T, K> {
+    T create(T product);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/GenericDao.java]:
```suggestion
    T create(T item);
```

File: src/main/java/com/internet/shop/dao/GenericDao.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+    Optional<T> getById(K item);
+
+    T update(T product);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/GenericDao.java]:
```suggestion
    T update(T item);
```

File: .travis.yml (hunk 3 above)
Comment (AndrewPolishchuc) [reply .travis.yml]: Done

File: src/main/java/com/internet/shop/dao/GenericDao.java (hunk 4 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/dao/GenericDao.java]: Done

File: src/main/java/com/internet/shop/dao/GenericDao.java (hunk 5 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/dao/GenericDao.java]: Done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**